BUG: Handle numpy (1.22+) error when attempting to rasterize string

https://github.com/corteva/geocube/runs/4707259171?

```
    def test_make_geocube__group_by_no_measurements(input_geodata, tmpdir):
>       out_grid = make_geocube(
            vector_data=input_geodata,
            output_crs=TEST_GARS_PROJ,
            geom=json.dumps(mapping(TEST_GARS_POLY)),
            group_by="hzdept_r",
            resolution=(-10, 10),
            fill=-9999.0,
        )

test/integration/api/test_core_integration.py:644: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
geocube/api/core.py:90: in make_geocube
    return VectorToCube(
geocube/vector_to_cube.py:163: in make_geocube
    return self._get_dataset(
geocube/vector_to_cube.py:233: in _get_dataset
    grid_array = self._get_grouped_grid(
geocube/vector_to_cube.py:294: in _get_grouped_grid
    image = self._rasterize_function(
geocube/rasterize.py:101: in rasterize_image
    image = rasterio.features.rasterize(
/usr/share/miniconda/envs/test/lib/python3.8/site-packages/rasterio/env.py:387: in wrapper
    return f(*args, **kwds)
/usr/share/miniconda/envs/test/lib/python3.8/site-packages/rasterio/features.py:325: in rasterize
    if not validate_dtype(shape_values, valid_dtypes):
/usr/share/miniconda/envs/test/lib/python3.8/site-packages/rasterio/dtypes.py:186: in validate_dtype
    get_minimum_dtype(values) in valid_dtypes)
/usr/share/miniconda/envs/test/lib/python3.8/site-packages/rasterio/dtypes.py:109: in get_minimum_dtype
    min_value = values.min()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = array(['12577452', '12577266', '12577459', '12577323', '12577225',
       '12577369', '12577275'], dtype='<U8')
axis = None, out = None, keepdims = False, initial = <no value>, where = True

    def _amin(a, axis=None, out=None, keepdims=False,
              initial=_NoValue, where=True):
>       return umr_minimum(a, axis, None, out, keepdims, initial, where)
E       numpy.core._exceptions._UFuncNoLoopError: ufunc 'minimum' did not contain a loop with signature matching types (dtype('<U8'), dtype('<U8')) -> None
```

--- a/docs/history.rst
+++ b/docs/history.rst
@@ -3,7 +3,7 @@ History
 
 Latest
 -------
-
+- BUG: Updated logic to skip rasterizing strings for numpy 1.22+ (issue #88)
 
 0.1.0
 ------

--- a/geocube/rasterize.py
+++ b/geocube/rasterize.py
@@ -8,7 +8,12 @@ import rasterio.features
 from rasterio.enums import MergeAlg
 from scipy.interpolate import Rbf, griddata
 
-from geocube.logger import get_logger
+
+def _is_numeric(data_values):
+    """
+    Check if array data type is numeric.
+    """
+    return numpy.issubdtype(data_values.dtype.type, numpy.number)
 
 
 def _remove_missing_data(data_values, geometry_array):
@@ -91,28 +96,23 @@ def rasterize_image(
         The vector data in the rasterized format.
 
     """
-    logger = get_logger()
+    if not _is_numeric(data_values):
+        # only numbers can be rasterized
+        return None
 
-    try:
-        if filter_nan:
-            data_values, geometry_array = _remove_missing_data(
-                data_values, geometry_array
-            )
-        image = rasterio.features.rasterize(
-            zip(geometry_array.values, data_values),
-            out_shape=(geobox.height, geobox.width),
-            transform=geobox.affine,
-            fill=fill,
-            all_touched=all_touched,
-            merge_alg=merge_alg,
-            dtype=_minimize_dtype(data_values.dtype, fill),
-        )
-        return image
-    except TypeError as ter:
-        if "cannot perform reduce with flexible type" in str(ter):
-            logger.warning(f"{ter}")
-            return None
-        raise
+    if filter_nan:
+        data_values, geometry_array = _remove_missing_data(data_values, geometry_array)
+
+    image = rasterio.features.rasterize(
+        zip(geometry_array.values, data_values),
+        out_shape=(geobox.height, geobox.width),
+        transform=geobox.affine,
+        fill=fill,
+        all_touched=all_touched,
+        merge_alg=merge_alg,
+        dtype=_minimize_dtype(data_values.dtype, fill),
+    )
+    return image
 
 
 def rasterize_points_griddata(
@@ -155,25 +155,21 @@ def rasterize_points_griddata(
     :class:`numpy.ndarray`: An interpolated :class:`numpy.ndarray`.
 
     """
-    if data_values.dtype == object:
+    if not _is_numeric(data_values):
+        # only numbers can be rasterized
         return None
-    try:
-        if filter_nan:
-            data_values, geometry_array = _remove_missing_data(
-                data_values, geometry_array
-            )
-        return griddata(
-            points=(geometry_array.x, geometry_array.y),
-            values=data_values,
-            xi=tuple(numpy.meshgrid(grid_coords["x"], grid_coords["y"])),
-            method=method,
-            fill_value=fill,
-            rescale=rescale,
-        )
-    except ValueError as ver:
-        if "could not convert string to float" in str(ver):
-            return None
-        raise
+
+    if filter_nan:
+        data_values, geometry_array = _remove_missing_data(data_values, geometry_array)
+
+    return griddata(
+        points=(geometry_array.x, geometry_array.y),
+        values=data_values,
+        xi=tuple(numpy.meshgrid(grid_coords["x"], grid_coords["y"])),
+        method=method,
+        fill_value=fill,
+        rescale=rescale,
+    )
 
 
 def rasterize_points_radial(
@@ -212,17 +208,12 @@ def rasterize_points_radial(
     :class:`numpy.ndarray`: An interpolated :class:`numpy.ndarray`.
 
     """
-    logger = get_logger()
+    if not _is_numeric(data_values):
+        # only numbers can be rasterized
+        return None
 
-    try:
-        if filter_nan:
-            data_values, geometry_array = _remove_missing_data(
-                data_values, geometry_array
-            )
-        interp = Rbf(geometry_array.x, geometry_array.y, data_values, function=method)
-        return interp(*numpy.meshgrid(grid_coords["x"], grid_coords["y"]))
-    except ValueError as ter:
-        if "object arrays are not supported" in str(ter):
-            logger.warning(f"{ter}")
-            return None
-        raise
+    if filter_nan:
+        data_values, geometry_array = _remove_missing_data(data_values, geometry_array)
+
+    interp = Rbf(geometry_array.x, geometry_array.y, data_values, function=method)
+    return interp(*numpy.meshgrid(grid_coords["x"], grid_coords["y"]))

